chemprop/v2/data/utils.py

```diff
@@ -1,66 +1,173 @@
-from collections import defaultdict
-from typing import Sequence
+import copy
+import itertools
+import logging
+from enum import auto
+from typing import Sequence, Tuple
 
 import numpy as np
-from rdkit.Chem.Scaffolds import MurckoScaffold
-
+from astartes import train_test_split, train_val_test_split
+from astartes.molecules import train_test_split_molecules, train_val_test_split_molecules
 
+from chemprop.v2.cli.utils import LOG_DIR, NOW
 from chemprop.v2.data.datapoints import MoleculeDatapoint
+from chemprop.v2.data.datasets import MoleculeDataset
+from chemprop.v2.data.utils import log_scaffold_stats
```

Comment:
This is importing from the file it is in (v2/data/utils.py). There is not `log_scaffold_stats` in this file. From what I can tell, in v1 there was data/scaffold.py which had this function. It should be ported over, perhaps even into astartes, if it is an important part of doing scaffold splits. If it is ported over into chemprop v2 (and not into astartes) we will want the function to not require a logger as it did in v1, but instead to do logging at the module level (as described in another comment).

Reply:
All this function did in v1 was print how many scaffolds were found, as well as the number per scaffold, and this comment reminded me that `astartes` actually does that already. Removing.

Reply:
Where does astartes print the scaffold information to? If possible we would like to send it to the logging file instead of std.out

Reply:
I should have looked before saying but `astartes` actually only prints out some limited warning information in the case that scaffolds aren't found for a molecule.

Reply:
A question, are we redirecting log output for all our external calls? I think we could do that globally, if so.